### PR TITLE
feat: hasMissingUsdcWallet logic to payout info form

### PR DESCRIPTION
**Pull request by kaelsky**
- [x] Payout required now depends on formData and `missingUsdcWallet` to accept either Ether or FIAT
- [x] Correctly handle plural on "networks" in statuses

### CODE REVIEW

File: src/pages/ProjectDetails/index.tsx [hunk 1]
```diff
@@ -31,7 +30,7 @@ export default function ProjectDetails() {
     return <ErrorFallback />;
   }
 
-  const { name } = data as Project;
+  const { name } = data as components["schemas"]["ProjectResponse"];
```
Comment (haydencleary) [inline src/pages/ProjectDetails/index.tsx:33]: If you move this line after you check the existence of `data` you don't need to cast the type.

File: src/App/Layout/Header/ProfileButton/PayoutInfoSidePanel/components/StatusTag.tsx [hunk 2]
```diff
@@ -25,23 +25,28 @@ const networksMessages = {
...
   const networks = requiredNetworks
     ? Object.entries(requiredNetworks)
         .filter(([, value]) => value)
-        .map(([key]) => T(networksMessages[key as keyof typeof networksMessages]))
-        .join(", ")
+        .map(([key]) =>
+          isFiat && key === "missingUsdcWallet"
+            ? T(networksMessages["missingSepaAccount"])
+            : T(networksMessages[key as keyof typeof networksMessages])
+        )
```
Comment (haydencleary) [inline src/App/Layout/Header/ProfileButton/PayoutInfoSidePanel/components/StatusTag.tsx:49]: There was a `join` here previously, is it no longer needed ?

File: src/App/Layout/Header/ProfileButton/PayoutInfoSidePanel/components/FormFields/OtherCryptoFields.tsx [hunk 3]
```diff
@@ -9,13 +9,23 @@ import { APTOS_WALLET, ETH_WALLET_OR_ENS_ADDRESS, OPTIMISM_WALLET, STARKNET_WALL
...
   const { T } = useIntl();
   const { register } = useFormContext();
 
+  const { watch } = useFormContext();
+  const [usdPreferredMethod, profileType] = watch(["usdPreferredMethod", "profileType"]);
+
+  const { missingAptosWallet, missingEthWallet, missingOptimismWallet, missingStarknetWallet, missingUsdcWallet } =
+    requiredFields || {};
+
+  const isEhterRequired =
```
Comment (haydencleary) [inline src/App/Layout/Header/ProfileButton/PayoutInfoSidePanel/components/FormFields/OtherCryptoFields.tsx:25]: Misspelled `Ether` I think

File: src/App/Layout/Header/ProfileButton/PayoutInfoSidePanel/components/FormFields/OtherCryptoFields.tsx [hunk 4]
```diff
@@ -9,13 +9,23 @@ import { APTOS_WALLET, ETH_WALLET_OR_ENS_ADDRESS, OPTIMISM_WALLET, STARKNET_WALL
...
+import { ProfileType } from "src/App/Layout/Header/ProfileButton/PayoutInfoSidePanel/types";
+import { PreferredMethod } from "src/types";
 
 export function OtherCryptoFields({ requiredFields }: { requiredFields: RequiredFieldsType }) {
-  const { missingAptosWallet, missingEthWallet, missingOptimismWallet, missingStarknetWallet } = requiredFields || {};
-
   const { T } = useIntl();
   const { register } = useFormContext();
 
+  const { watch } = useFormContext();
```
Comment (haydencleary) [inline src/App/Layout/Header/ProfileButton/PayoutInfoSidePanel/components/FormFields/OtherCryptoFields.tsx:19]: Couldn't this be extracted from the `useFormContext` above ?

File: src/pages/ProjectDetails/index.tsx (hunk 1 above)
Comment (kaelsky) [reply src/pages/ProjectDetails/index.tsx:33]: Directly used L67, so we can use `networks.length` as a `count` value

File: src/App/Layout/Header/ProfileButton/PayoutInfoSidePanel/components/StatusTag.tsx (hunk 2 above)
Comment (kaelsky) [reply src/App/Layout/Header/ProfileButton/PayoutInfoSidePanel/components/StatusTag.tsx:49]: Used at line L67 so we can maintain `networks` as an array for the count

**Review by haydencleary — APPROVED** [review]

**Pull request merged**